build.gradle

```diff
@@ -1,71 +1,115 @@
-    jvm_version = '1.8'
+    ver = [
+        'kotlin'     : '1.3.31',
+        'jackson'    : '2.9.9',
+        'junit'      : '5.4.2',
+        'cloudEvents': 'b8b9c6fef3',
+    ]
+
+    servicePackage = 'org.openbroker'
 }
 
-apply from: "jar.gradle"
+sourceCompatibility = jdkVersion
+targetCompatibility = jdkVersion
 
 repositories {
+    maven {
+        url "https://jitpack.io"
+        content {
+            includeGroupByRegex 'com\\.(github|gitlab|gitee).*'
+        }
+    }
     jcenter()
     mavenCentral()
-    maven { url "https://dl.bintray.com/kotlin/kotlinx" }
-    maven { url "https://jitpack.io" }
 }
 
 dependencies {
-    // Kotlin
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 
-    // Cloud Events
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37'
-    compile 'com.github.open-broker:cloud-events-jvm:b8b9c6fef37:sources'
+    implementation(
+        platform("org.jetbrains.kotlin:kotlin-bom:$ver.kotlin"),
+        platform("com.fasterxml.jackson:jackson-bom:$ver.jackson"),
+        "org.jetbrains.kotlin:kotlin-stdlib-jdk8",
+        "com.fasterxml.jackson.core:jackson-core",
+        "com.fasterxml.jackson.module:jackson-module-kotlin",
+        "com.fasterxml.jackson.datatype:jackson-datatype-jsr310",
+        "com.github.open-broker:cloud-events-jvm:$ver.cloudEvents",
+    )
 
-    // Jackson
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version:'2.9.8'
+    testImplementation(
+        platform("org.junit:junit-bom:$ver.junit"),
+        "org.junit.jupiter:junit-jupiter-api",
+    )
 
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.3.1")
-    testRuntime("org.junit.jupiter:junit-jupiter-engine:5.3.1")
+    testRuntime(
+        "org.junit.jupiter:junit-jupiter-engine",
+    )
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
     kotlinOptions {
-        languageVersion = kotlin_api_version
-        apiVersion = kotlin_api_version
-        jvmTarget = jvm_version
-        javaParameters = true   // Useful for reflection.
+        javaParameters = true
+        jvmTarget = jdkVersion
+        freeCompilerArgs += ['-Xjsr305=strict', '-progressive']
+        languageVersion = kotlinLanguageVersion
+        apiVersion = kotlinLanguageVersion
     }
 }
 
-compileJava {
-    sourceCompatibility = jvm_version
-    targetCompatibility = jvm_version
-    options.incremental = true
-    options.encoding = 'UTF-8'
+tasks.withType(JavaCompile).all {
+    options.encoding = defaultEncoding
+    options.compilerArgs = ['-parameters', '-Xlint:unchecked']
+}
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    archiveClassifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+publishing {
```

Comment:
At this stage when we have not set up artifactory what does this task add?

Reply:
Sorry, I do not understand why you are talking about the artifactory. You use jitpack, this is also a maven repository. Publishing is a task that describes how it will be published, as you see, you can specify which artifacts we need to publish, if necessary, you can customize the POM file created.

In general, this is a standard gradle plugin that is used to publish any artifacts. JitPack uses it when building

Reply:
There was already a build job that has been working successfully with JitPack. If you replace the existing build job with another way of building, it has to be clear *why* such replacement is an improvement. We are not saying the change is bad, but maybe you can explain why this is a better alternative?

Reply:
The main thing why it better: because maven plugin is deprecated (from gradle version 1.3) and in gradle 4.x gradle team finilized maven-publish plugin and recommend upgrade scripts to new maven-publish plugin. You can read about these plugins here:

https://docs.gradle.org/current/userguide/maven_plugin.html

https://docs.gradle.org/current/userguide/publishing_maven.html

From my side I also recommend use new maven-publish plugin because it's more powerfull, convenient and flexible

Reply:
Okay, then I am satisfies as long as it gives same functionality as previous plugin :)

Reply:
It gives you a lot more ;-)